src/components/uri/Visualization.tsx

```diff
@@ -306,6 +306,7 @@ export class VisualizationWrapped extends React.Component<
             this.subject.next(
                 this.prepareDataSources(nextProps.projectId, nextProps.identifier, nextProps.filters),
             );
+            this.setState({ error: null });
```

Comment:
Why do we reset the error here? Is `hasInvalidDatasource` not an error state?

Reply:
The thing with hasInvalidDatasource is that it is checking whether props has been changed, in the code right now, we do not reset the error state when props has been changed; therefore, visualization is not re-rendering correctly and it is still rendering the Error component.

Reply:
OK, thanks for the explanation :)